daceml/onnx/op_implementations/fpga_implementations.py

```diff
@@ -886,6 +886,1034 @@ def make_compute(sdfg, state, vec_width=1):
+                schedule=dace.ScheduleType.FPGA_Device,
+                unroll=True)
+
+            state.add_memlet_path(
+                vec_buf_data,
+                vector_out_entry,
+                write_pipe_task,
+                dst_conn="buf",
+                memlet=dace.Memlet(
+                    f"vec_buf_data[((({store_local_index_img}) - (int_floor({store_local_index_img}, {vec_width_in}) * {vec_width_in})) + m1)]",
+                    dynamic=True,
+                ))
+
+            state.add_memlet_path(b_dummy,
+                                  map_entry,
+                                  vector_out_entry,
+                                  write_pipe_task,
+                                  dst_conn="dummy_value",
+                                  memlet=dace.Memlet("B_dummy[0]"))
+
+            state.add_memlet_path(write_pipe_task,
+                                  vector_out_exit,
+                                  vec_buf_out,
+                                  src_conn="to_kernel",
+                                  memlet=dace.Memlet("vec_buf_out[m1]", ))
+
+            # connect graph reader and writer sections
+            state.add_memlet_path(dummy_container,
+                                  vector_out_entry,
+                                  write_pipe_task,
+                                  dst_conn="dummy_con",
+                                  memlet=dace.Memlet("dummy_container[0]"))
+
+            # ----------------------------------------
+            # write to output vector buffer
+            # ----------------------------------------
+            write_out_task = state.add_tasklet(
+                "write_out", {"in_con": dace.vector(base_type, vec_width_in)},
+                {"out_con"},
+                f"if (not ({load_test})) and (m0 < {T}/{vec_width}): out_con = in_con"
+            )
+
+            state.add_memlet_path(vec_buf_out,
+                                  write_out_task,
+                                  dst_conn="in_con",
+                                  memlet=dace.Memlet("vec_buf_out"))
+
+            state.add_memlet_path(write_out_task,
+                                  map_exit,
+                                  pipe,
+                                  src_conn="out_con",
+                                  memlet=dace.Memlet("B_pipe[0]",
+                                                     dynamic=True))
+
+        def make_write_C(state, add_bias=True):
+            # Receives the results and adds it to C
+            # i.e. receives results, adds Bias input B and outputs into Y
+
+            vendor = dace.config.Config.get("compiler", "fpga_vendor")
+            xilinx = (vendor == "xilinx")
+
+            pipe = state.add_read("C_pipe")
+            if add_bias:
+                mem_read = state.add_read("B")
+            mem = state.add_write("Y")
+
+            # Because compilers, I don't know
```

Comment:
Leftover?

Reply:
I removed the cynical comment :) but the rest still applies, that is why we have `if xilinx:` and only then we pre-fetch bias values.